hypha/apply/dashboard/static/js/django_select2-checkboxes.js

```diff
@@ -1,23 +1,34 @@
-(function ($) {
-    var init = function ($element) {
-        options = {
-            placeholder: $element.data('placeholder'),
-            templateSelection: function(selected, total) {
-                let filterType = this.placeholder;
-                if ( !selected.length ) {
-                    return filterType;
-                } else if ( selected.length===total ) {
-                    return 'All ' + filterType + ' selected';
-                }
-                return selected.length + ' of ' + total + ' ' + filterType + ' selected';
-            }
-        };
-        $element.select2MultiCheckboxes(options);
-    };
-    $(function () {
+(function($) {
+  $.fn.select2.amd.require(
+    [
+      'select2/multi-checkboxes/dropdown',
+      'select2/multi-checkboxes/selection',
+      'select2/multi-checkboxes/results'
+    ],
+    function(DropdownAdapter, SelectionAdapter, ResultsAdapter) {
+
+      $(function () {
         $('.django-select2-checkboxes').each(function (i, element) {
-            var $element = $(element);
-            init($element);
+          var $element = $(element);
+
+          $element.select2({
+            placeholder: $element.data('placeholder'),
+            templateSelection: function(data) {
```

Comment:
Tried to add `closeOnSelect: false,` and it seems to work but then I get some errors when closing the popup.

`TypeError: undefined is not an object (evaluating 'r.GetData(this,"element").select2')` in `select2.min.js`

`TypeError: undefined is not an object (evaluating 'match[0].classList')` in `/apply/submission-filters.js:41`

Reply:
Let me check this.

Reply:
yes `closeOnSelect` works but give errors on the console.
I tried looking for what exactly the issue is, this is what I have got but that didn't seem to help as there is no resolution on it.
 https://github.com/select2/select2/issues/3282
https://github.com/select2/select2/issues/3800
I will spend some time more and see if I can have a workaround, I tried the following. 
```
          if ($element !== undefined) {
            $element.select2({
              placeholder: $element.data('placeholder'),
              closeOnSelect: false,
              templateSelection: function(data) {
                let filterType = $element.data('placeholder');

                if (!data.selected.length) {
                  return filterType
                } else if (data.selected.length == data.all.length) {
                  return 'All ' + filterType + ' selected';
                }
                return data.selected.length + ' of ' + data.all.length + ' ' + filterType + ' selected';
              },
              selectionAdapter: SelectionAdapter,
              returnesultsAdapter: ResultsAdapter
            });
          };
```

Reply:
@frjo 

The thing is jQuery and $ were available when it applies select2 checkbox plugin for the first time but not afterward, 

There are solutions like changing sequence of jquery and other plugin used in HTML file, but that breaks more things.